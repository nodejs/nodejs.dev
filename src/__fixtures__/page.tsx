import {
  PaginationInfo,
  LearnTemplateContext,
  NavigationData,
  PostTemplateData,
  PostTemplateContext,
  NodeReleaseData,
  PageTableOfContents,
  BlogCategory,
  BlogPost,
  ArticleData,
} from '../types';
import mockMDXBodyContent from './mockMDXBodyContent';

export const mockTableOfContents: PageTableOfContents = {
  items: [
    { title: 'mock-title', url: '#mock-title' },
    { title: 'mock-title-2', url: '#mock-title-2' },
  ],
};

export const createPaginationInfo = (): PaginationInfo =>
  ({
    slug: 'test-slug',
    title: 'test-title',
  } as PaginationInfo);

export const createNavigationSectionData = (): NavigationData =>
  ({
    'test-section': [
      {
        slug: 'test-slug-1',
        title: 'test-title-1',
      },
      {
        title: 'test-title-2',
        slug: 'test-slug-2',
      },
    ],
    'test-section2': [
      {
        slug: 'test-slug-3',
        title: 'test-title-3',
      },
      {
        title: 'test-title-4',
        slug: 'test-slug-4',
      },
    ],
  } as NavigationData);

export const createLearnPageData = () => ({
  data: {
    articleCurrentLanguage: {
      body: mockMDXBodyContent,
      tableOfContents: mockTableOfContents,
      frontmatter: {
        title: 'test-title',
        description: 'test-description',
      },
      fields: {
        authors: ['test-user1', 'test-user2'],
      },
    } as ArticleData,
  },
});

export const createNodeReleasesData = (): NodeReleaseData[] =>
  [
    {
      endOfLife: '2022-04-30',
      ltsStart: '2019-10-21',
      maintenanceStart: '2020-11-30',
      initialRelease: '2019-04-23',
      codename: 'erbium',
      version: 'v12',
      fullVersion: 'v12.0.0',
      status: 'Maintenance LTS',
      isLts: false,
    },
    {
      endOfLife: '2023-04-30',
      ltsStart: '2020-10-17',
      maintenanceStart: '2021-10-19',
      initialRelease: '2020-04-21',
      codename: 'fermium',
      version: 'v14',
      fullVersion: 'v14.0.0',
      status: 'Active LTS',
      isLts: true,
    },
    {
      endOfLife: '2025-04-30',
      ltsStart: '2022-10-25',
      maintenanceStart: '2023-10-18',
      initialRelease: '2022-04-19',
      codename: 'v18',
      version: 'v18',
      fullVersion: 'v18.0.0',
      status: 'Current',
      isLts: false,
    },
  ] as NodeReleaseData[];

export const createLearnPageContext = (): LearnTemplateContext =>
  ({
    slug: 'test-slug',
    relativePath: 'test-path',
    next: createPaginationInfo(),
    previous: createPaginationInfo(),
    navigationData: createNavigationSectionData(),
  } as LearnTemplateContext);

export const createBlogPageContext = (): PostTemplateContext => ({
  slug: 'test-slug',
  relativePath: 'test-path',
  next: createPaginationInfo(),
  previous: createPaginationInfo(),
  navigationData: createNavigationSectionData(),
  recent: [
    {
      node: {
        frontmatter: {
          title: 'title-mock',
          category: {
            name: 'category-mock',
            slug: 'category-mock-slug',
          },
          blogAuthors: [
            {
              id: 'id-mock',
              name: 'name-mock',
              website: 'url-mock',
            },
          ],
        },
        fields: {
          date: '11/11/2022',
          slug: 'slug-mock',
          readingTime: { text: 'text-mock' },
        },
      },
    },
  ],
});

export const createBlogData = (): {
  posts: { edges: BlogPost[] };
  categories: { edges: BlogCategory[] };
} => ({
  posts: {
    edges: [
      {
        node: {
          frontmatter: {
            title: 'Mock blog title',
            category: {
              name: 'mock-category',
              slug: 'Mock Category Slug',
            },
            blogAuthors: [
              {
                name: 'Mock author name',
                website: 'Mock URL',
              },
            ],
          },
          fields: {
            date: '11/11/2022',
            slug: 'Mock blog slug',
            readingTime: {
              text: '1 min read',
            },
          },
        },
      },
    ],
  },
  categories: {
    edges: [],
  },
});

export const createBlogPageData = (): PostTemplateData => ({
  mdx: {
    body: mockMDXBodyContent,
    excerpt: 'excerpt-mock',
    frontmatter: {
      title: 't,itle-mock',
      blogAuthors: [
        {
          id: 'id-mock',
          name: 'name-mock',
          website: 'url-mock',
        },
      ],
    },
    fields: { slug: 'slug-mock', date: '11/11/2022' },
  },
});

export const createGeneralPageData = () => ({
  data: {
    articleCurrentLanguage: {
      fields: {
        authors: ['author-mock'],
      },
      frontmatter: {
        description: 'Mock Description',
        title: 'Mock Title',
      },
      body: mockMDXBodyContent,
      tableOfContents: mockTableOfContents,
    } as ArticleData,
  },
});

export const createResourcesData = () => ({
  data: {
    articleCurrentLanguage: {
      fields: {
        authors: ['MrJithil'],
      },
      frontmatter: {
        description: 'Mock Description',
        title: 'Mock Title',
      },
      body: mockMDXBodyContent,
      tableOfContents: mockTableOfContents,
    } as ArticleData,
  },
});

export const createPrivacyData = () => ({
  data: {
    articleCurrentLanguage: {
      fields: {
        authors: ['Author'],
      },
      frontmatter: {
        description: 'Description',
        title: 'Title',
      },
      body: mockMDXBodyContent,
      tableOfContents: mockTableOfContents,
    } as ArticleData,
  },
});
