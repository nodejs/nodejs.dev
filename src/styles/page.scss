.page > .article-reader {
  margin: 0 auto;
}
