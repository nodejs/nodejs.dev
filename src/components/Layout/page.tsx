import React from 'react';
import Article from '../Article';
import Layout from '.';

import '../../styles/page.scss';

interface Page {
  editPath?: string;
  html: string;
  tableOfContents: string;
  title: string;
  description: string;
  authors: string[];
  location?: Location;
}

export default function PageLayout(props: Page): JSX.Element {
  const {
    editPath,
    html,
    tableOfContents,
    title,
    description,
    authors,
    location,
  } = props;
  return (
    <Layout title={title} location={location} description={description}>
      <main className="page">
        <Article
          html={html}
          title={title}
          authors={authors}
          editPath={editPath}
          tableOfContents={tableOfContents}
        />
      </main>
    </Layout>
  );
}
