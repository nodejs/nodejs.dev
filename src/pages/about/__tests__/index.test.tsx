import React from 'react';
import { render } from '@testing-library/react';
import AboutPage from '..';
import { createGeneralPageData } from '../../../__fixtures__/page';

const mockData = createGeneralPageData();

describe('About page', () => {
  it('renders correctly', () => {
    const { container } = render(<AboutPage data={mockData.data} />);
    // eslint-disable-next-line testing-library/no-container, testing-library/no-node-access
    const pageContent = container.querySelector('main');

    expect(pageContent).toMatchSnapshot();
  });
});
