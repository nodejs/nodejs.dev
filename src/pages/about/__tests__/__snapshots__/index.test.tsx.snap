// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About page renders correctly 1`] = `
<main
  class="grid-container"
>
  <nav
    aria-label="About Navigation"
    class="navigation"
    id="main-navigation"
  >
    <button
      aria-expanded="false"
      class="navigationOpen"
      type="button"
    >
      Menu
    </button>
    <div
      class="navigationSection"
    >
      <button
        aria-expanded="true"
        class="t-body2 title"
        type="button"
      >
        <span>
          About Node.js
        </span>
      </button>
      <div
        aria-label="About Node.js"
        role="region"
        style="display: block;"
      >
        <a
          aria-current="page"
          class="t-body2 navigationItem navigationItemActive"
          href="/about/"
          id="/about/"
        >
          About Node.js
        </a>
        <a
          class="t-body2 navigationItem"
          href="/about/governance/"
          id="/about/governance/"
        >
          Project Governance
        </a>
        <a
          class="t-body2 navigationItem"
          href="/about/releases/"
          id="/about/releases/"
        >
          Releases
        </a>
        <a
          class="t-body2 navigationItem"
          href="/about/resources/"
          id="/about/resources/"
        >
          Resources
        </a>
        <a
          class="t-body2 navigationItem"
          href="https://privacy-policy.openjsf.org/"
          id="https://privacy-policy.openjsf.org/"
        >
          Privacy Policy
        </a>
        <a
          class="t-body2 navigationItem"
          href="/about/security/"
          id="/about/security/"
        >
          Security Reporting
        </a>
      </div>
    </div>
    <div
      class="navigationSection"
    >
      <button
        aria-expanded="true"
        class="t-body2 title"
        type="button"
      >
        <span>
          Get Involved
        </span>
      </button>
      <div
        aria-label="Get Involved"
        role="region"
        style="display: block;"
      >
        <a
          class="t-body2 navigationItem"
          href="/get-involved/"
          id="/get-involved/"
        >
          Get Involved
        </a>
        <a
          class="t-body2 navigationItem"
          href="/get-involved/collab-summit"
          id="/get-involved/collab-summit"
        >
          Collab Summit
        </a>
        <a
          class="t-body2 navigationItem"
          href="/get-involved/contribute"
          id="/get-involved/contribute"
        >
          Contribute
        </a>
        <a
          class="t-body2 navigationItem"
          href="https://github.com/nodejs/node/blob/main/doc/contributing/code-of-conduct.md"
          id="https://github.com/nodejs/node/blob/main/doc/contributing/code-of-conduct.md"
        >
          Code of Conduct
        </a>
      </div>
    </div>
    <div
      class="navigationSection"
    >
      <button
        aria-expanded="true"
        class="t-body2 title"
        type="button"
      >
        <span>
          Download
        </span>
      </button>
      <div
        aria-label="Download"
        role="region"
        style="display: block;"
      >
        <a
          class="t-body2 navigationItem"
          href="/download/"
          id="/download/"
        >
          Download
        </a>
        <a
          class="t-body2 navigationItem"
          href="/download/package-manager/"
          id="/download/package-manager/"
        >
          Package Manager
        </a>
        <a
          class="t-body2 navigationItem"
          href="/download/releases/"
          id="/download/releases/"
        >
          Previous Releases
        </a>
      </div>
    </div>
  </nav>
  <article
    class="article"
  >
    <div>
      <h1
        class="headline"
      >
        Mock Title
      </h1>
      <p />
    </div>
    <div />
    <div />
    <div>
      <p>
        Node.js can be installed in different ways. This post highlights the most common and convenient ones.
      </p>
      <p>
        Official packages for all the major platforms are available at 
        <a>
          https://nodejs.dev/download/
        </a>
        .
      </p>
      <p>
        One very convenient way to install Node.js is through a package manager. In this case, every operating system has its own.
      </p>
      <p>
        Other package managers for MacOS, Linux, and Windows are listed in 
        <a>
          https://nodejs.dev/download/package-manager/
        </a>
      </p>
      <p>
        <code
          class="code"
        >
          nvm
        </code>
         is a popular way to run Node.js. It allows you to easily switch the Node.js version, and install new versions to try and easily rollback if something breaks, for example.
      </p>
      <p>
        It is also very useful to test your code with old Node.js versions.
      </p>
      <p>
        See 
        <a>
          https://github.com/nvm-sh/nvm
        </a>
         for more information about this option.
      </p>
      <p>
        In any case, when Node.js is installed you'll have access to the 
        <code
          class="code"
        >
          node
        </code>
         executable program in the command line.
      </p>
    </div>
    <div
      class="edit"
    >
      <a
        href="https://github.com/nodejs/nodejs.dev/edit/main/content/about/about.en.md"
      >
        <span>
          Edit this page on GitHub
        </span>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-pencil "
          data-icon="pencil"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M410.3 231l11.3-11.3-33.9-33.9-62.1-62.1L291.7 89.8l-11.3 11.3-22.6 22.6L58.6 322.9c-10.4 10.4-18 23.3-22.2 37.4L1 480.7c-2.5 8.4-.2 17.5 6.1 23.7s15.3 8.5 23.7 6.1l120.3-35.4c14.1-4.2 27-11.8 37.4-22.2L387.7 253.7 410.3 231zM160 399.4l-9.1 22.7c-4 3.1-8.5 5.4-13.3 6.9L59.4 452l23-78.1c1.4-4.9 3.8-9.4 6.9-13.3l22.7-9.1v32c0 8.8 7.2 16 16 16h32zM362.7 18.7L348.3 33.2 325.7 55.8 314.3 67.1l33.9 33.9 62.1 62.1 33.9 33.9 11.3-11.3 22.6-22.6 14.5-14.5c25-25 25-65.5 0-90.5L453.3 18.7c-25-25-65.5-25-90.5 0zm-47.4 168l-144 144c-6.2 6.2-16.4 6.2-22.6 0s-6.2-16.4 0-22.6l144-144c6.2-6.2 16.4-6.2 22.6 0s6.2 16.4 0 22.6z"
            fill="currentColor"
          />
        </svg>
      </a>
    </div>
  </article>
</main>
`;
